src/Modals/AssignRoute.js

```diff
@@ -0,0 +1,147 @@
+import { ButtonLoading, PrimaryButton } from '@components/Button.js';
+import { axiosBase as axios } from '@utils/Api.js';
+import React, { useState, useEffect } from 'react';
+import { Link, useLocation, useNavigate } from 'react-router-dom';
+import { toast } from 'react-toastify';
+import Swal from 'sweetalert2';
+
+export const assignRouteToBus = async (data) => {
+  try {
+    await axios.post(`/bus-to-routes/${data.plateNumber}/${data.route}`);
+    return true;
+  } catch(error) {
+    toast('Something went wrong on our end');
+  }
+};
+
+export const AssignRouteModal = ({ handleClose }) => {
+  const navigate = useNavigate();
+  const [routes, setRoutes] = useState()
+  const [loading, setloading] = useState(false);
+  const location = useLocation();
+  const { plateNumber, busType, seats, route, id } = location?.state;
+  const onSubmit = async (e) => {
+    e.preventDefault();
+    const data = {
+      plateNumber: plateNumber,
+      route: e.target.route.value
+    }
+    try {
+      setloading(true);
+      console.log(route)
+      if (route) {
+        Swal.fire({
+          title: 'Are you sure?',
+          text: 'This bus already have a route!',
+          icon: 'warning',
+          showCancelButton: true,
+          confirmButtonColor: '#3085d6',
+          cancelButtonColor: '#d33',
+          confirmButtonText: 'Yes, Assign route!'
+        }).then(async (result) => {
+          if (result.isConfirmed) {
+            await assignRouteToBus(data);
+            Swal.fire(
+              'Done!',
+              'The bus has been assigned a new route.',
+              'success'
+            ).then(() => {
+              navigate('/dashboard/management');
+            });
+          }
+        });
+      } else {
+        await assignRouteToBus(data);
+        toast('The bus has been assigned a new route.', { type: 'success' });
+        navigate('/dashboard/management');
+      }
+
+    } catch (error) {
+      console.log(error)
```

Comment:
This log is no longer needed and remember to check if there is an error from the backend instead of using the generic error as we are integrating.
We can try `error?.response?.data?.data?.message || error.message`